onnxscript/rewriter/pattern.py

```diff
@@ -504,13 +585,35 @@ def commute(self) -> list[ValuePattern]:
         return [self]
 
 
+def _nodes_in_pattern(outputs: Sequence[ValuePattern | None]) -> list[NodePattern]:
+    """Computes inputs and nodes from outputs of a pattern."""
+    node_patterns: list[NodePattern] = []
+
+    def visit(value_patterns: Sequence[ValuePattern]) -> None:
+        for value_pattern in value_patterns:
+            if isinstance(value_pattern, NodeOutputPattern):
+                node_pattern = value_pattern.producer()
+                if node_pattern not in node_patterns:
+                    node_patterns.append(node_pattern)
+                    visit(node_pattern.inputs)
```

Comment:
Would there be stack overflow?

Reply:
No cycles are possible. Pattern lengths are not expected to be extremely long. Are you recommending a worklist based algorithm?